File: app/src/main/java/tech/salroid/filmy/parser/FavouriteMovieParseWork.java
package tech.salroid.filmy.parser;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

import tech.salroid.filmy.data.FavouriteData;

public class FavouriteMovieParseWork {

    private String result;

    public FavouriteMovieParseWork(String result) {
        this.result = result;
    }

    public List<FavouriteData> parseFavourite() {
        final List<FavouriteData> favouriteArray = new ArrayList<>();
        FavouriteData favouriteData;

        try {

            JSONObject jsonobject = new JSONObject(result);
            JSONArray jsonArray = jsonobject.getJSONArray("results");

            for (int i = 0; i < jsonArray.length(); i++) {

                String title, id, movie_poster;

                id = jsonArray.getJSONObject(i).getString("id");
                movie_poster = "http://image.tmdb.org/t/p/w185" + jsonArray.getJSONObject(i).getString("poster_path");
                title = jsonArray.getJSONObject(i).getString("original_title");

                favouriteData = new FavouriteData(id);

                if (!(movie_poster.equals("null"))) {
                    favouriteData.setId(id);
                    favouriteData.setTitle(title);
                    favouriteData.setPoster(movie_poster);
                    favouriteArray.add(favouriteData);
                }

            }
        } catch (JSONException e) {
            e.printStackTrace();
        }

        return favouriteArray;
    }
}

File: app/src/main/java/tech/salroid/filmy/parser/FavouriteMovieParseWork.kt
/*
package tech.salroid.filmy.parser

import org.json.JSONException
import org.json.JSONObject
import tech.salroid.filmy.data.FavouriteData
import java.util.*

class FavouriteMovieParseWork(private val result: String) {

    fun parseFavourite(): List<FavouriteData> {
        val favouriteArray: MutableList<FavouriteData> = ArrayList()
        var favouriteData: FavouriteData

        try {
            val jsonObject = JSONObject(result)
            val jsonArray = jsonObject.getJSONArray("results")

            for (i in 0 until jsonArray.length()) {
                val id: String = jsonArray.getJSONObject(i).getString("id")
                val poster: String = "http://image.tmdb.org/t/p/w185" + jsonArray.getJSONObject(i)
                    .getString("poster_path")
                val title: String? = jsonArray.getJSONObject(i).getString("original_title")
                favouriteData = FavouriteData(id)

                if (poster != "null") {
                    favouriteData.title = title
                    favouriteData.poster = poster
                    favouriteArray.add(favouriteData)
                }
            }
        } catch (e: JSONException) {
            e.printStackTrace()
        }
        return favouriteArray
    }
}*/
